### PR TITLE
[Refactor] Add delay for camera status update

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
## Purpose

* Added delay for camera switching and for turning camera on
* Updated Changelog

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

## What to Check
Verify that the following are valid
* turn on-of camera quickly a couple of times, a local camera video should be shown as expected
* switch between back and front camera quickly a couple of times, a local camera video should be shown as expected

### CODE REVIEW

**Review by AmyL219 — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Redux/Middleware/CallingMiddlewareHandler.swift [hunk 1]
```diff
@@ -199,9 +199,10 @@ class CallingMiddlewareHandler: CallingMiddlewareHandling {
             dispatch(PermissionAction.CameraPermissionRequested())
         } else {
             callingService.startLocalVideoStream()
+                .delay(for: 1.0, scheduler: DispatchQueue.main)
```
Comment (JoshuaLai) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Redux/Middleware/CallingMiddlewareHandler.swift:202]: The documentation on this is not very clear. I am not sure if it is seconds or milliseconds. 
Can we make this more clear by changing it to `.seconds(1)` or `.milliseconds(1)` whatever the case maybe

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 ### Bugs Fixed
 - Fixed issue where header was still selectable with voiceover on and overlay visible. [#256](https://github.com/Azure/communication-ui-library-ios/pull/256)
 - Fixed hold overlay to have a solid colour. [#262](https://github.com/Azure/communication-ui-library-ios/pull/262) 
-- Fix the issue that resume a call without internet could stop user from exit. [#268](https://github.com/Azure/communication-ui-library-ios/pull/268
+- Fixed the issue that resume a call without internet could stop user from exit. [#268](https://github.com/Azure/communication-ui-library-ios/pull/268
```
Comment (JoshuaLai) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/CHANGELOG.md:7]: Lets add a change log entry for this

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/CallingViewComponent/ControlBarViewModel.swift [hunk 3]
```diff
@@ -122,7 +128,8 @@ class ControlBarViewModel: ObservableObject {
     }
 
     func isCameraDisabled() -> Bool {
-        cameraPermission == .denied || cameraState.operation == .pending || callingStatus == .localHold
+        cameraPermission == .denied || cameraState.operation == .pending ||
+        callingStatus == .localHold || isCameraStateUpdating
```
[the remark this answers is not in the document]
Comment (AmyL219) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/CallingViewComponent/ControlBarViewModel.swift:132]: We would exit the callComposite when .remoteHold happens. For current implementation, it's not needed. After the CallingSDK fixed the bug, the `.remoteHold` wouldn't appear in group call. I can leave .remoteHold not handled. Until one day we have that 1:1 call scenario

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/CallingViewComponent/ControlBarViewModel.swift [hunk 4]
```diff
@@ -180,6 +187,9 @@ class ControlBarViewModel: ObservableObject {
         }
 
         cameraState = localUserState.cameraState
+        if isCameraStateUpdating {
```
[the remark this answers is not in the document]
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/CallingViewComponent/ControlBarViewModel.swift]: Added additional check to handle this edge case

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Redux/Middleware/CallingMiddlewareHandler.swift (hunk 1 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Redux/Middleware/CallingMiddlewareHandler.swift]: Updated!

**Review by JoshuaLai — APPROVED** [review]

**Review by AmyL219 — APPROVED** [review]

**Pull request merged**